internal/pkg/service/buffer/storage/local/writer/csv/csv_benchmark_test.go

```diff
@@ -0,0 +1,265 @@
+				wb.Compression = compression.DefaultZSTDConfig()
+				wb.Compression.ZSTD.Level = 1
+				wb.Sync.Mode = disksync.ModeDisk
+				wb.Sync.Wait = true
+			},
+		},
+		{
+			Name: "compression=ZSTD_SpeedFastest,sync=ToDisk,wait=false",
+			Configure: func(wb *benchmark.WriterBenchmark) {
+				wb.Compression = compression.DefaultZSTDConfig()
+				wb.Compression.ZSTD.Level = zstd.SpeedFastest
+				wb.Sync.Mode = disksync.ModeDisk
+				wb.Sync.Wait = false
+			},
+		},
+		{
+			Name: "compression=ZSTD_SpeedDefault,sync=ToDisk,wait=true",
+			Configure: func(wb *benchmark.WriterBenchmark) {
+				wb.Compression = compression.DefaultZSTDConfig()
+				wb.Compression.ZSTD.Level = zstd.SpeedDefault
+				wb.Sync.Mode = disksync.ModeDisk
+				wb.Sync.Wait = true
+			},
+		},
+		{
+			Name: "compression=ZSTD_SpeedDefault,sync=ToDisk,wait=false",
+			Configure: func(wb *benchmark.WriterBenchmark) {
+				wb.Compression = compression.DefaultZSTDConfig()
+				wb.Compression.ZSTD.Level = zstd.SpeedDefault
+				wb.Sync.Mode = disksync.ModeDisk
+				wb.Sync.Wait = false
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		b.Run(tc.Name, func(b *testing.B) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			wb := newBenchmark(b)
+			if tc.Configure != nil {
+				tc.Configure(wb)
+			}
+			wb.Run(ctx)
+		})
+	}
+}
+
+func newBenchmark(b *testing.B) *benchmark.WriterBenchmark {
+	columns := column.Columns{
+		// 8 columns, only the count is important for CSV
+		column.ID{},
+		column.Datetime{},
+		column.Body{},
+		column.Template{},
+		column.Template{},
+		column.Template{},
+		column.Template{},
+		column.Template{},
+	}
+
+	return &benchmark.WriterBenchmark{
+		B:           b,
+		Parallelism: 10000,
+		FileType:    storage.FileTypeCSV,
+		Columns:     columns,
+		Allocate:    100 * datasize.MB,
+		Sync: disksync.Config{
+			Mode:            disksync.ModeDisk,
+			Wait:            true,
+			CheckInterval:   5 * time.Millisecond,
+			CountTrigger:    500,
+			BytesTrigger:    1 * datasize.MB,
+			IntervalTrigger: 50 * time.Millisecond,
+		},
+		Compression: compression.DefaultNoneConfig(),
+		DataChFactory: func(ctx context.Context, n int, g *benchmark.RandomStringGenerator) <-chan []any {
+			ch := make(chan []any, 1000)
+			columnsCount := len(columns)
+			columnLength := int(benchmarkRowLength.Bytes()) / columnsCount
+
+			// Pre-generate unique rows
+			rows := make([][]any, benchmarkUniqueRows)
+			for i := 0; i < benchmarkUniqueRows; i++ {
+				rows[i] = make([]any, columnsCount)
+				for j := 0; j < len(columns); j++ {
+					rows[i][j] = g.RandomString(columnLength)
+				}
+			}
+
+			// Send the pre-generated rows to the channel over and over
+			go func() {
+				defer close(ch)
+				row := 0
+				for i := 0; i < n; i++ {
+					if ctx.Err() != nil {
+						break
+					}
+					ch <- rows[row]
+					row++
+					if row == benchmarkUniqueRows {
+						row = 0
+					}
+				}
+			}()
```

Comment:
A potom zapisujeme do `ch`, pokial nedosiahneme `n`.

`n` si urci Go benchmak sam, napr. povies, ze benchmark ma bezat `-benchtime=2s`.
- Spusti sa jedna iteracia a odhadne sa `n` aby to bezalo `2s`.

Nas zaujima skor konstantny pocet zapisov, preto pouziva `-benchtime=10000x` resp `100000x` ked chcem presnejsie vysledky.

Nizsi pocet iteracii je pomalsi, je zatazeny setupovanim writers a inicializovaním vsetkych buffers co tam su.